operator/video/v0/video_operation.go

```diff
@@ -0,0 +1,185 @@
+package video
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/instill-ai/component/base"
+	ffmpeg "github.com/u2takey/ffmpeg-go"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type SubsampleVideoInput struct {
+	Video     Video  `json:"video"`
+	Fps       int    `json:"fps"`
+	StartTime string `json:"start-time"`
+	Duration  string `json:"duration"`
+}
+
+type SubsampleVideoOutput struct {
+	Video Video `json:"video"`
+}
+
+type SubsampleVideoFramesInput struct {
+	Video     Video  `json:"video"`
+	Fps       int    `json:"fps"`
+	StartTime string `json:"start-time"`
+	Duration  string `json:"duration"`
+}
+
+type SubsampleVideoFramesOutput struct {
+	Frames []Frame `json:"frames"`
+}
+
+// Base64 encoded video
+type Video string
+
+// Base64 encoded frame
+type Frame string
+
+func subsampleVideo(input *structpb.Struct) (*structpb.Struct, error) {
+
+	inputStruct := SubsampleVideoInput{}
+
+	err := base.ConvertFromStructpb(input, &inputStruct)
+	if err != nil {
+		return nil, fmt.Errorf("error converting input to struct: %v", err)
+	}
+
+	base64Video := string(inputStruct.Video)
+
+	videoBytes, err := base64.StdEncoding.DecodeString(base.TrimBase64Mime(base64Video))
+
+	if err != nil {
+		return nil, fmt.Errorf("error in decoding for inner: %s", err)
+	}
+
+	videoPrefix := strings.Split(base64Video, ",")[0]
+
+	// TODO: chuang8511 map the file extension to the correct format
+	// using mp4 first because it has smaller size.
+	tempFileIn := "temp.mp4"
+
+	err = os.WriteFile(tempFileIn, videoBytes, 0644)
```

Comment:
We should avoid using constant names for storing video files, as it could cause issues if we receive two requests simultaneously.
Maybe we can try using https://pkg.go.dev/os#CreateTemp to create a tempfile.

Reply:
fix.
For frame one, I do not use CreateTemp because 1 run will generate the multiple frames.
It means we cannot know how many frames it will generate in advance.
To fix it, I just set uuid as prefix to make it avoid duplicate.